nix/pkgs/xcodeenv/compose-xcodewrapper.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, lib, writeShellScriptBin }:
+{ versions ? [ "15.1" ]
+, xcodeBaseDir ? "/Applications/Xcode.app" }:
+
+assert stdenv.isDarwin;
+
+let
+  xcodebuildPath = "${xcodeBaseDir}/Contents/Developer/usr/bin/xcodebuild";
+
+  xcodebuildWrapper = writeShellScriptBin "xcodebuild" ''
+    currentVer="$(${xcodebuildPath} -version | awk 'NR==1{print $2}')"
+    wrapperVers=(${lib.concatStringsSep " " versions})
+
+    for ver in "''${wrapperVers[@]}"; do
+      if [[ "$currentVer" == "$ver" ]]; then
+        exec "${xcodebuildPath}" "$@"
+        exit 0
+      fi
+    done
```

Comment:
I would invert the logic to put the `exec` ad the end of the script.

Also, `exit 0` is not necessary since `exec` means the command provided takes over execution of the script:

>If command is supplied, it replaces the shell without creating a new process. 

https://www.gnu.org/software/bash/manual/bash.html#index-exec

Reply:
I agree on the removal of `exit 0` but not with inverting the logic.
since `versions` is a list, I think its better if we execute `xcodebuild` as soon as we find that the local `Xcode` version is matching any of the versions provided to `xcodewrapper`.  
The list might grow in future.. at the moment we are only compatible with Xcode 15.1 and hence you see only 1 item passed as a param to `versions`